sync/sync.go

```diff
@@ -180,6 +188,48 @@ func (s *Synchronizer) fetchUnknownClasses(ctx context.Context, stateUpdate *cor
 	return newClasses, closer()
 }
 
+func (s *Synchronizer) handlePluginRevertBlock() {
+	if s.plugin == nil {
+		return
+	}
+	fromBlock, err := s.blockchain.Head()
+	if err != nil {
+		s.log.Warnw("Failed to retrieve the reverted blockchain head block for the plugin", "err", err)
+		return
+	}
+	fromSU, err := s.blockchain.StateUpdateByNumber(fromBlock.Number)
+	if err != nil {
+		s.log.Warnw("Failed to retrieve the reverted blockchain head state-update for the plugin", "err", err)
+		return
+	}
+	reverseStateDiff, err := s.blockchain.GetReverseStateDiff()
+	if err != nil {
+		s.log.Warnw("Failed to retrieve reverse state diff", "head", fromBlock.Number, "hash", fromBlock.Hash.ShortString(), "err", err)
+		return
+	}
+	var toBlock *core.Block
+	var toSU *core.StateUpdate
+	if fromBlock.Number != 0 {
+		toBlock, err = s.blockchain.BlockByHash(fromBlock.ParentHash)
+		if err != nil {
+			s.log.Warnw("Failed to retrieve the parent block for the plugin", "err", err)
+			return
+		}
+		toSU, err = s.blockchain.StateUpdateByNumber(toBlock.Number)
+		if err != nil {
+			s.log.Warnw("Failed to retrieve the parents state-update for the plugin", "err", err)
+			return
+		}
+	}
+	err = (s.plugin).RevertBlock(
+		&junoplugin.BlockAndStateUpdate{Block: fromBlock, StateUpdate: fromSU},
+		&junoplugin.BlockAndStateUpdate{Block: toBlock, StateUpdate: toSU},
```

Comment:
maybe in case of `blockNumber == 0` it's better to pass nil explicitly here? because structure in this case won't have a lot of value (both values will be nil)